test/cli_run_memory_test.go

```diff
@@ -35,73 +35,88 @@ func (suite *PouchRunMemorySuite) TearDownTest(c *check.C) {
 
 // TestRunWithMemoryswap is to verify the valid running container
 // with --memory-swap
-func (suite *PouchRunMemorySuite) TestRunWithMemoryswap(c *check.C) {
+func (suite *PouchRunMemorySuite) TestRunWithMemoryswapAndKernelMemory(c *check.C) {
 	SkipIfFalse(c, environment.IsMemorySupport)
 	SkipIfFalse(c, environment.IsMemorySwapSupport)
 
-	cname := "TestRunWithMemoryswap"
-	m := 1024 * 1024
+	cname := "TestRunWithMemoryswapAndKernelMemory"
 	memory := "100m"
 	memSwap := "200m"
-	expected := 200 * m
-	sleep := "10000"
+	kernelMemory := "100m"
+	expectedMem := strconv.Itoa(100 * 1024 * 1024)       // 100 MB
+	expectedKernelMem := strconv.Itoa(1024 * 1024 * 100) // 100 MB
+	expectedMemSwap := strconv.Itoa(200 * 1024 * 1024)   // 200 MB
 
-	res := command.PouchRun("run", "-d", "-m", memory,
+	res := command.PouchRun("run", "-d",
+		"--memory", memory,
 		"--memory-swap", memSwap,
-		"--name", cname, busyboxImage, "sleep", sleep)
+		"--kernel-memory", kernelMemory,
+		"--name", cname,
+		busyboxImage,
+		"sleep", "10000")
 	defer DelContainerForceMultyTime(c, cname)
 	res.Assert(c, icmd.Success)
 
 	// test if the value is in inspect result
-	memorySwap, err := inspectFilter(cname, ".HostConfig.MemorySwap")
+	memoryResult, err := inspectFilter(cname, ".HostConfig.Memory")
 	c.Assert(err, check.IsNil)
-	c.Assert(memorySwap, check.Equals, strconv.Itoa(expected))
+	c.Assert(memoryResult, check.Equals, expectedMem)
+	memorySwapResult, err := inspectFilter(cname, ".HostConfig.MemorySwap")
+	c.Assert(err, check.IsNil)
+	c.Assert(memorySwapResult, check.Equals, expectedMemSwap)
+	kernelMemoryResult, err := inspectFilter(cname, ".HostConfig.KernelMemory")
+	c.Assert(err, check.IsNil)
+	c.Assert(kernelMemoryResult, check.Equals, expectedKernelMem)
 
 	// test if cgroup has record the real value
 	containerID, err := inspectFilter(cname, ".ID")
 	c.Assert(err, check.IsNil)
-	path := fmt.Sprintf(
-		"/sys/fs/cgroup/memory/default/%s/memory.memsw.limit_in_bytes",
-		containerID)
-	checkFileContains(c, path, strconv.Itoa(expected))
+	path := fmt.Sprintf("/sys/fs/cgroup/memory/default/%s/memory.limit_in_bytes", containerID)
+	checkFileContains(c, path, expectedMem)
+	path = fmt.Sprintf("/sys/fs/cgroup/memory/default/%s/memory.memsw.limit_in_bytes", containerID)
+	checkFileContains(c, path, expectedMemSwap)
+	path = fmt.Sprintf("/sys/fs/cgroup/memory/default/%s/memory.kmem.usage_in_bytes", containerID)
+	checkFileContains(c, path, expectedKernelMem)
```

Comment:
something wrong here

```
FAIL: cli_run_memory_test.go:38: PouchRunMemorySuite.TestRunWithMemoryswapAndKernelMemory


cli_run_memory_test.go:79:

    checkFileContains(c, path, expectedKernelMem)

cli_run_test.go:248:
    c.Assert(strings.Contains(string(cmdResult.Stdout()), expt),
        check.Equals, true)
... obtained bool = false
... expected bool = true
```

Reply:
Oh, I should use `memory.kmem.limit_in_bytes` rather than `memory.kmem.usage_in_bytes`. Updated.